File: rocketmq-go/model/send_result.go
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package model

import (
	"fmt"
	"github.com/apache/incubator-rocketmq-externals/rocketmq-go/model/message"
)

type SendStatus int

const (
	SendOK SendStatus = iota
	//FlushDiskTimeout
	//FlushSlaveTimeout
	SlaveNotAvaliable
)

type SendResult struct {
	sendStatus    SendStatus
	msgID         string
	messageQueue  *message.MessageQueue
	queueOffset   int64
	transactionID string
	offsetMsgID   string
	regionID      string
	traceOn       bool
}

func NewSendResult(status SendStatus, msgID, offsetID string, queue *message.MessageQueue, queueOffset int64) *SendResult {
	return &SendResult{
		sendStatus:   status,
		msgID:        msgID,
		offsetMsgID:  offsetID,
		messageQueue: queue,
		queueOffset:  queueOffset,
	}
}

func EncoderSendResultToJson(obj interface{}) string {
	return "" // TODO
}

func DecoderSendResultFromJson(json string) *SendResult {
	return nil // TODO
}

func (result *SendResult) TraceOn() bool {
	return result.traceOn
}

func (result *SendResult) SetTraceOn(b bool) {
	result.traceOn = b
}

func (result *SendResult) SetRegionID(s string) {
	result.regionID = s
}

func (result *SendResult) MsgID() string {
	return result.msgID
}

func (result *SendResult) SetMsgID(s string) {
	result.msgID = s
}

func (result *SendResult) SendStatus() SendStatus {
	return result.sendStatus
}

func (result *SendResult) SetSendStatus(status SendStatus) {
	result.sendStatus = status
}

func (result *SendResult) MessageQueue() *message.MessageQueue {
	return result.messageQueue
}

func (result *SendResult) SetMessageQueue(queue *message.MessageQueue) {
	result.messageQueue = queue
}

func (result *SendResult) QueueOffset() int64 {
	return result.queueOffset
}

func (result *SendResult) SetQueueOffset(offset int64) {
	result.queueOffset = offset
}

func (result *SendResult) TransactionID() string {
	return result.transactionID
}

func (result *SendResult) SetTransactionID(s string) {
	result.transactionID = s
}

func (result *SendResult) OffsetMsgID() string {
	return result.offsetMsgID
}

func (result *SendResult) SetOffsetMsgID(s string) {
	result.offsetMsgID = s
}

func (result *SendResult) String() string {
	return fmt.Sprintf("SendResult [sendStatus=%s, msgId=%s, offsetMsgId=%s, messageQueue=%s, queueOffset=%s]",
		result.sendStatus, result.msgID, result.offsetMsgID, result.messageQueue, result.queueOffset)
}

File: rocketmq-cpp/src/producer/SendResult.cpp
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
#include "SendResult.h"
#include "UtilAll.h"
#include "VirtualEnvUtil.h"

namespace rocketmq {
//<!***************************************************************************
SendResult::SendResult() : m_sendStatus(SEND_OK), m_queueOffset(0) {}

SendResult::SendResult(const SendStatus& sendStatus, const string& msgId,
                       const MQMessageQueue& messageQueue, int64 queueOffset)
    : m_sendStatus(sendStatus),
      m_msgId(msgId),
      m_messageQueue(messageQueue),
      m_queueOffset(queueOffset) {}

SendResult::SendResult(const SendResult& other) {
  m_sendStatus = other.m_sendStatus;
  m_msgId = other.m_msgId;
  m_messageQueue = other.m_messageQueue;
  m_queueOffset = other.m_queueOffset;
}

SendResult& SendResult::operator=(const SendResult& other) {
  if (this != &other) {
    m_sendStatus = other.m_sendStatus;
    m_msgId = other.m_msgId;
    m_messageQueue = other.m_messageQueue;
    m_queueOffset = other.m_queueOffset;
  }
  return *this;
}

SendResult::~SendResult() {}

const string& SendResult::getMsgId() const { return m_msgId; }

SendStatus SendResult::getSendStatus() const { return m_sendStatus; }

MQMessageQueue SendResult::getMessageQueue() const { return m_messageQueue; }

int64 SendResult::getQueueOffset() const { return m_queueOffset; }

//<!************************************************************************
}  //<!end namespace;
